### PR TITLE
clippy: Fix a variety of clippy warnings in `fonts`, `layout_2020` and the DOM code

### PR DESCRIPTION
**Pull request by chickenleaf**
PR Description
Warnings and Fixes:
1. Unused import in `font.rs` - removed
2. complex type used in `style_ext.rs` - as discussed [here ](https://github.com/servo/servo/pull/33708), allowed the warnings
3. Allowed `enum_variant_names` in auto-generated rust code `PrototypeList.rs` through `CodegenRust.py`
4. too many arguments in `closeevent.rs` and `compositionevent.rs` - allowed
5. removed `'static` lifetime in `gstreamer.py`
6. Arc not Send and Sync in `refcounted.rs` - allowed as discussed [here](https://github.com/servo/servo/issues/33726)

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes are part of https://github.com/servo/servo/issues/31500

- [ ] There are tests for these changes OR
- [X] These changes do not require tests because they do not touch functionality.

### CODE REVIEW

File: components/script/dom/compositionevent.rs [hunk 1]
```diff
@@ -34,7 +34,7 @@ impl CompositionEvent {
     pub fn new_uninitialized(window: &Window) -> DomRoot<CompositionEvent> {
         reflect_dom_object(Box::new(CompositionEvent::new_inherited()), window)
     }
-
+    #[allow(clippy::too_many_arguments)]
```
Comment (mrobinson) [inline components/script/dom/compositionevent.rs:37]: Please preserve the newline between the two functions:
```suggestion

    #[allow(clippy::too_many_arguments)]
```

**Review by mrobinson — CHANGES_REQUESTED** [review]

File: components/script/dom/compositionevent.rs (hunk 1 above)
Comment (chickenleaf) [reply components/script/dom/compositionevent.rs:37]: done, have added the newline

File: components/script/dom/bindings/codegen/CodegenRust.py [hunk 2]
```diff
@@ -7935,6 +7935,7 @@ def PrototypeList(config):
             CGGeneric(AUTOGENERATED_WARNING_COMMENT),
             CGGeneric(f"pub const PROTO_OR_IFACE_LENGTH: usize = {len(protos) + len(constructors)};\n"),
             CGGeneric(f"pub const MAX_PROTO_CHAIN_LENGTH: usize = {config.maxProtoChainLength};\n\n"),
+            CGGeneric("#[allow(clippy::enum_variant_names)]\n"),
```
Comment (mukilan) [inline components/script/dom/bindings/codegen/CodegenRust.py:7938]:
```suggestion
            CGGeneric("#[allow(clippy::enum_variant_names)]"), 
```

The explicit newline (`\n`)  is not needed here. With this explicit newline, the generated `PrototypeList.rs` will have a blank line between this `allow` and the rest of the enum. Technically the extra newline doesn't change the sematics, but it does make the generated code look "wrong".

Comment (chickenleaf) [reply components/script/dom/bindings/codegen/CodegenRust.py:7938]: Understood! I shall fix this

**Review by mrobinson — APPROVED** [review]

**Comment by mrobinson** [comment]
Thank you!

**Review by mukilan — APPROVED** [review]

**Pull request merged**